core/controller/src/whisk/core/controller/Actions.scala

```diff
@@ -456,19 +475,19 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * either there is an error in the get, or the promise has completed because it timed out. The promise MUST
      * complete in the caller to terminate the polling.
      */
-    private def pollDbForResult(docid: DocInfo, activationId: ActivationId, resultOnly: Boolean,
-                                promise: Promise[(ActivationId, Option[(ActivationResponse, JsObject)])])(implicit transid: TransactionId): Unit = {
+    private def pollDbForResult(
+        docid: DocInfo,
+        activationId: ActivationId,
+        promise: Promise[Option[WhiskActivation]])(
+            implicit transid: TransactionId): Unit = {
         if (!promise.isCompleted) {
             WhiskActivation.get(activationStore, docid) map {
-                fullActivation =>
-                    val activation = fullActivation.stripLogs
```

Comment:
why striplogs? if getting the activation from the db, logs already exist or might exist, and might as well pass them on in the response to the client if providing the full activation back.